File: algorithms/0002/main.js
const { ListNode } = require('../../kit/ListNode')
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
  const answer = new ListNode()
  let copy = answer
  let tmp = 0
  while (l1 || l2) {
    const a = l1 ? l1.val : 0
    const b = l2 ? l2.val : 0
    const sum = a + b + tmp
    tmp = Math.floor(sum / 10)
    copy.next = new ListNode(sum % 10)
    copy = copy.next
    if (l1) {
      l1 = l1.next
    }
    if (l2) {
      l2 = l2.next
    }
  }
  if (tmp) {
    copy.next = new ListNode(tmp)
  }
  return answer.next
}

module.exports = addTwoNumbers

File: algorithms/0002/main.go
package problem0002

// ListNode Definition for singly-linked list
type ListNode struct {
	Val  int
	Next *ListNode
}

func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
	answer := &ListNode{}
	current := answer
	tmp := 0

	for l1 != nil || l2 != nil {
		sum := tmp
		if l1 != nil {
			sum += l1.Val
			l1 = l1.Next
		}
		if l2 != nil {
			sum += l2.Val
			l2 = l2.Next
		}
		tmp = sum / 10
		current.Next = &ListNode{Val: sum % 10}
		current = current.Next
	}
	if tmp != 0 {
		current.Next = &ListNode{Val: tmp}
	}

	return answer.Next
}
